pkg/istio/controller/virtual_services.go

```diff
@@ -0,0 +1,23 @@
+package controller
+
+import "github.com/knative/pkg/apis/istio/v1alpha3"
+import "github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
+
+// GetVirtualServicesForGateway returns the VirtualServices for the provided gateway
+func GetVirtualServicesForGateway(gateway v1alpha3.Gateway, c *k8scontext.Context) []*v1alpha3.VirtualService {
```

Comment:
Can you please move code related to istio resource getting/listing/filtering to k8sContext package. See if you can use the handlers to populate the cache instead of post processing.